src/components/MainApp/index.tsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import { Form } from 'react-bootstrap';
+import { InputNewTodo } from '../InputNewTodo';
+import UserSelect from '../UserSelect';
+import { connect } from 'react-redux';
+import styles from './MainApp.module.css';
+
+
+type Todo = {
+    title: string,
+    user?: number,
+    isDone: boolean,
+}
+
+type MainAppProps = {
+    todos: Todo[],
+    addTodo: (t: Todo) => void,
+    changeTodo: (todos: Todo[]) => void,
+}
+type MainAppState = {
+    todoTitle: string
+};
+
+class Index extends React.Component<MainAppProps, MainAppState> {
+    constructor(props: MainAppProps) {
+        super(props);
+        this.state = { todoTitle: '' }
+    }
```

Comment:
I would refactor this component into a functional component and I would use hooks like:
`useState` - to manipulate state
`useSelector` & ``useDispatch - to communicate with `redux` (instead of using a class component and `connect` from `redux`